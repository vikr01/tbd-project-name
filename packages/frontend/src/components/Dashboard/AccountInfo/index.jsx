// @flow
import React, { Fragment, Component } from 'react';
import axios from 'axios';
import { Typography, Paper } from '@material-ui/core';
import { Link } from 'react-router-dom';
import routes from '../../../routes';

export default class AccountInfoView extends Component {
  state = {
    name: '',
    username: '',
    creditInfo: '',
    error: false,
    loaded: false,
  };

  async componentDidMount() {
    await this.populateState();
  }

  async populateState() {
    let response;
    try {
      response = await axios.get(backendRoutes.USER);
    } catch (error) {
      console.error(error);
      this.setState({ error: true, loaded: true });
      return;
    }
    console.log(response);
    this.setState({
      name: `${response.data.firstName} ${response.data.lastName}`,
      username: response.data.username,
      creditInfo: response.data.creditCard,
      accountType: response.data.accountType,
      seatNumber: response.data.seatNumber,
      error: false,
      loaded: true,
    });
  }

  renderContent() {
    const {
      error,
      loaded,
      name,
      username,
      creditInfo,
      accountType,
      seatNumber,
    } = this.state;
    if (!loaded) {
      return (
        <Typography variant="h4" gutterBottom component="h2">
          Loading...
        </Typography>
      );
    }
    if (error) {
      return (
        <Typography variant="h4" gutterBottom component="h2">
          Unable to load user info
        </Typography>
      );
    }
    return (
      <Fragment>
        <Typography variant="h4" gutterBottom component="h2">
          Account details
        </Typography>
        <div>
          <Typography variant="h5" className="accountOverviewItem">
            {`Name: ${name}`}
          </Typography>
          <Typography variant="h5" className="accountOverviewItem">
            {`Username: ${username}`}
          </Typography>
          {accountType === 'Client' && (
            <Fragment>
              <Typography variant="h5" className="accountOverviewItem">
                {`Card number: ${creditInfo}`}
              </Typography>
              <Link to={routes.CREDITCARD_ADD} className="accountOverviewItem">
                Enter Credit card info
              </Link>
            </Fragment>
          )}
          {accountType === 'Driver' && (
            <Fragment>
              <Typography variant="h5" className="accountOverviewItem">
                {`Seat count: ${seatNumber}`}
              </Typography>
              <Link to={routes.CARSEATS_SET} className="accountOverviewItem">
                Change seat count
              </Link>
            </Fragment>
          )}
        </div>
      </Fragment>
    );
  }

  render() {
    return (
      <Fragment>
        <Paper className="paperDashboard">{this.renderContent()}</Paper>
        <div className="rest" />
      </Fragment>
    );
  }
}
