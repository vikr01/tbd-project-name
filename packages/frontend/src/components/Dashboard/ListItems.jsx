// @flow
import React, { Fragment } from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ExitToApp from '@material-ui/icons/';
import PeopleIcon from '@material-ui/icons/People';
import SvgIcon from '@material-ui/core/SvgIcon';

type Props = {
  onClick: func,
};

const ListItems = ({ onClick }: Props) => (
  <Fragment>
    <ListItem
      button
      style={{ width: 'auto' }}
      onClick={() => onClick('Dashboard')}
    >
      <ListItemIcon>
        <SvgIcon style={{ color: 'white' }}>
          <path d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z" />
        </SvgIcon>
      </ListItemIcon>
      <ListItemText
        primaryTypographyProps={{ variant: 'inherit' }}
        primary="Dashboard"
      />
    </ListItem>
    <ListItem
      button
      style={{ width: 'auto' }}
      onClick={() => onClick('Account')}
    >
      <ListItemIcon>
        <PeopleIcon style={{ color: 'white' }} />
      </ListItemIcon>
      <ListItemText
        primaryTypographyProps={{ variant: 'inherit' }}
        primary="Account information"
      />
    </ListItem>
    <ListItem
      button
      style={{ width: 'auto' }}
      onClick={() => onClick('Logout')}
    >
      <ListItemIcon>
        <SvgIcon style={{ color: 'white' }}>
          <path d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" />
        </SvgIcon>
      </ListItemIcon>
      <ListItemText
        primaryTypographyProps={{ variant: 'inherit' }}
        primary="Logout"
      />
    </ListItem>
  </Fragment>
);

export default ListItems;
