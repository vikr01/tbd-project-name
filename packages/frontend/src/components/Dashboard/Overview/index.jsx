// @flow
import React, { Fragment } from 'react';
import { Paper } from '@material-ui/core';
import MapView from './MapView';

class OverviewView extends React.Component {
  state = { showMap: false };

  doRequest() {
    this.setState({ showMap: true });
  }

  render() {
    const { showMap } = this.state;
    return (
      <Fragment>
        <Paper className="paperDashboard">
          <MapView showMap={showMap} startRequest={() => this.doRequest()} />
        </Paper>
      </Fragment>
    );
  }
}

export default OverviewView;
