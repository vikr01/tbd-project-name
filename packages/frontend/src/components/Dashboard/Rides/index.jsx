// @flow
import React, { Fragment } from 'react';

import { Typography, Paper } from '@material-ui/core';

const RidesView = () => (
  <Fragment>
    <Paper className="paperDashboard">
      <Typography variant="display1" gutterBottom component="h2">
        Rides
      </Typography>
    </Paper>
    <div className="rest" />
  </Fragment>
);

export default RidesView;
