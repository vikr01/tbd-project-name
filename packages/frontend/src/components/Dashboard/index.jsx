// @flow
import React, { Component, Fragment } from 'react';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';

import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import NotificationsIcon from '@material-ui/icons/Notifications';
import ListItems from './ListItems';
import RidesView from './Rides';
import AccountInfoView from './AccountInfo';
import OverviewView from './Overview';

const drawerWidth = 240;

const styles = theme => ({
  root: {
    display: 'flex',
  },
  toolbar: {
    paddingRight: 24, // keep right padding when drawer closed
  },
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginLeft: 12,
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    flexGrow: 1,
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing.unit * 7,
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing.unit * 9,
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    height: '100vh',
    overflow: 'auto',
  },
  chartContainer: {
    marginLeft: -22,
  },
  tableContainer: {
    height: 320,
  },
});

const NofiticationBadge = () => {
  const count = 0; // make state if we choose to add notifications
  if (count !== 0) {
    return (
      <Badge badgeContent={count} color="secondary">
        <NotificationsIcon />
      </Badge>
    );
  }
  return <NotificationsIcon />;
};

type Props = {
  classes: Object,
};

class Dashboard extends Component<Props> {
  props: Props;

  state = {
    open: true,
    page: 'Dashboard',
  };

  handleDrawerOpen = () => {
    this.setState({ open: true });
  };

  handleDrawerClose = () => {
    this.setState({ open: false });
  };

  handleMenuClick = e => {
    this.setState({ page: e });
  };

  render() {
    const { classes } = this.props;
    const { open, page } = this.state;
    return (
      <Fragment>
        <CssBaseline />
        <div className={classes.root}>
          <AppBar
            position="absolute"
            className={classNames(classes.appBar, open && classes.appBarShift)}
          >
            <Toolbar disableGutters={!open} className={classes.toolbar}>
              <IconButton
                color="inherit"
                aria-label="Open drawer"
                onClick={this.handleDrawerOpen}
                className={classNames(
                  classes.menuButton,
                  open && classes.menuButtonHidden
                )}
              >
                <MenuIcon />
              </IconButton>
              <Typography
                component="h1"
                variant="title"
                color="inherit"
                noWrap
                className={classes.title}
              >
                Dashboard
              </Typography>
              <IconButton color="inherit">
                <NofiticationBadge />
              </IconButton>
            </Toolbar>
          </AppBar>
          <Drawer
            variant="permanent"
            classes={{
              paper: classNames(
                classes.drawerPaper,
                !open && classes.drawerPaperClose
              ),
            }}
            open={open}
          >
            <div className={classes.toolbarIcon}>
              <IconButton onClick={this.handleDrawerClose}>
                <ChevronLeftIcon />
              </IconButton>
            </div>
            <Divider />
            <List>
              <ListItems
                onClick={e => {
                  this.handleMenuClick(e);
                }}
              />
            </List>
          </Drawer>
          <ShowMainContent page={page} classes={classes} />
        </div>
      </Fragment>
    );
  }
}

const ShowMainContent = ({ page, classes }: props) => {
  if (page === 'Rides') {
    return <RidesView classes={classes} />;
  }
  if (page === 'Account') {
    return <AccountInfoView classes={classes} />;
  }
  if (page === 'Dashboard') {
    return <OverviewView classes={classes} />;
  }
  return null;
};

export default withStyles(styles)(Dashboard);
