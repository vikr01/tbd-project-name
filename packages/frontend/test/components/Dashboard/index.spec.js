import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from 'toms-shuttles-frontend/src/components/Dashboard';

describe('Dashboard', () => {
  test('snapshot matches', () => {
    const dashboard = <Dashboard />;
    const tree = renderer.create(dashboard);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
