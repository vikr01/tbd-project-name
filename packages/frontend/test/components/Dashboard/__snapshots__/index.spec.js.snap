// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot matches 1`] = `
<p
  className="MuiTypography-root-12 MuiTypography-body1-21"
>
  Logging in...
</p>
`;
